### DISCUSSION TITLE
Best Practices, Using the Mesh.cs to generate objects.

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#731] [Q&A]
I need to generate a model from a collection of objects. As an example, say I want to "trace" a real-world object by placing world anchors at points about the object outline. From these points I could gather the information necessary to "connect the dots". To keep the example simple, say the object is comprised of two boxes placed together so it looks like a school bus.
I know I could simply use the generate cube method to make this simple example. But suppose it is not exactly cubic in nature and I want to "trace" the outline and use that to generate a single sidewall of the bus. How would I go about this?
In the Mesh.cs file documentation it mentions that the methods "generate a new mesh on the GPU". Is that a bad thing? If it is, what is a good solution?

**Comment by laultman** [comment]
@maluoi @laultman Hi Nick. I've done a lot of research and have a somewhat working model for object generation from a point cloud using the mesh object. I likely have some misunderstandings around the generation of the mesh vertices. In your Geo demo you have a "wavy" mesh object. If I had two triangles that shared an edge where triangle 1 has points a,b,c and triangle 2 has points a,b,d. If wanted to create a mesh from these two triangles where the two triangles are at right angles along the shared edge, how would I code the vertices for such an object. I want the two triangles to have their faces to the outside.
I've been trying to create a mesh using the new Vertex defining the three values of the vertex. The direction Vector doesn't seem to do anything that I can see.
I would appreciate a little assistance here in understanding the mesh construction.

**Reply by maluoi** [reply]
I'd recommend Catlike Coding's materials. They're focused on Unity, but StereoKit's procedural mesh API is pretty similar and the fundamentals are the same.

https://catlikecoding.com/unity/tutorials/procedural-meshes/

**Comment by laultman** [comment]
Maybe I don't understand the mesh at all. Is a cube a collection of meshes? I certainly can create a collection of mesh objects and draw them on the screen but that seems rather expensive especially if the mesh surfaces are fairly complex with lots of faces.

**Reply by maluoi** [reply]
A `Model` at its simplest can be thought of as a collection of `Mesh`/`Material` pairs. A cube could be a single `Mesh`, or a collection of `Mesh`es, depending on how it's modeled. It could be many meshes if it has different `Material`s applied to different faces. Typically it's just a single `Mesh` though.
